crates/biome_js_analyze/src/lint/nursery/use_google_font_preconnect.rs

```diff
@@ -0,0 +1,126 @@
+
+use crate::JsRuleAction;
+
+declare_lint_rule! {
+    /// Ensure `preconnect` is used with Google Fonts.
+    ///
+    /// When using Google Fonts, adding the `rel="preconnect"` attribute to the `<link>` tag
+    /// that points to `https://fonts.gstatic.com` is recommended to initiate an early
+    /// connection to the font's origin. This improves page load performance by reducing latency.
+    ///
+    /// Failing to use `preconnect` may result in slower font loading times, affecting user experience.
+    ///
+    /// Note: Next.js automatically adds this preconnect link starting from version 12.0.1, but in cases
+    /// where it's manually added, this rule ensures the `preconnect` attribute is properly used.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link rel="preload" href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <link rel="preconnect" href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ```jsx
+    /// <link href="/logo.svg" rel="icon" />
+    /// ```
+    ///
+    pub UseGoogleFontPreconnect {
+        version: "next",
+        name: "useGoogleFontPreconnect",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("google-font-preconnect")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for UseGoogleFontPreconnect {
+    type Query = Ast<AnyJsxElement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if node.name().ok()?.name_value_token()?.text_trimmed() != "link" {
+            return None;
+        }
+
+        if node
+            .get_attribute_inner_string_text("href")
+            .map_or(false, |href| href.starts_with("https://fonts.gstatic.com"))
+            && node
+                .get_attribute_inner_string_text("rel")
+                .map_or(true, |rel| rel != "preconnect")
+        {
+            return Some(node.syntax().text_range());
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        return Some(RuleDiagnostic::new(
+            rule_category!(),
+            range,
+            markup! { ""<Emphasis>"rel=\"preconnect\""</Emphasis>" is missing from Google Font." },
+        ));
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+
+        if node.find_attribute_by_name("rel").is_some() {
+            return None;
+        }
+
+        let mut mutation = ctx.root().begin();
+        let mut attributes: Vec<_> = node.attributes().into_iter().collect();
```

Comment:
```suggestion
        let mut attributes: Vec<_> = node.attributes().iter().collect();
```

Internally, both `iter` and `into_iter` yield the same result.